### ISSUE TITLE
Configure Fails On File Not Found -- Sun Workshop Compiler and    64-bit Support

### ISSUE THREAD

**Issue by p5pRT** [Perl/perl5#2102]
Migrated from [rt.perl.org#3392](https://rt-archive.perl.org/perl5/Ticket/Display.html?id=3392) (status was 'resolved')

Searchable as RT3392$

**Comment by p5pRT** [comment]
### From garry@ifr.inside.zvolve.net

#### Created by garry@zvolve.com



Configure fails when selecting 64\-bit support and using the Sun 
Workshop compiler&#8203;: 

There is an error in hints/solaris\_2\.sh that generates a \`dot' command
that specifies the wrong path for a file\.  It's on line 417\.  Here is
a patch that fixes the problem&#8203;: 

&nbsp;    \[exper\] /home/garry/src/perl\-5\.6\.0 $ diff \-c ~/xx hints/solaris\_2\.sh
&nbsp;    \*\*\* /home/garry/xx      Sat Jun 17 10&#8203;:41&#8203;:30 2000
&nbsp;    \-\-\- hints/solaris\_2\.sh  Sat Jun 17 10&#8203;:41&#8203;:48 2000
&nbsp;    \*\*\*\*\*\*\*\*\*\*\*\*\*\*\*
&nbsp;    \*\*\* 414\,421 \*\*\*\*
&nbsp;      EOM
&nbsp;            exit 1
&nbsp;        fi 
&nbsp;    \! \#         \. \./UU/workshoplibpth\.cbu
&nbsp;    \!           \. \./workshoplibpth\.cbu
&nbsp;        case "$cc \-v 2>/dev/null" in
&nbsp;        \*gcc\*\)
&nbsp;            echo 'main\(\) \{ return 0; \}' > try\.c
&nbsp;    \-\-\- 414\,420 \-\-\-\-
&nbsp;      EOM
&nbsp;            exit 1
&nbsp;        fi 
&nbsp;    \!           \. \./UU/workshoplibpth\.cbu
&nbsp;        case "$cc \-v 2>/dev/null" in
&nbsp;        \*gcc\*\)
&nbsp;            echo 'main\(\) \{ return 0; \}' > try\.c
&nbsp;    \[exper\] /home/garry/src/perl\-5\.6\.0 $ 

I used the following options to install&#8203;: 

&nbsp;    mkdir ~/perl ~/perl/bin

&nbsp;    sh Configure \-Dprefix=/home/garry/perl

&nbsp;    Directories to use for library searches?
&nbsp;    \[/usr/local/lib /opt/SUNWspro/SC5\.0/lib /lib /usr/lib /usr/ccs/lib\] /usr/lib/sparcv9 /usr/local/lib

&nbsp;    Any additional cc flags? \[\-I/usr/local/include\] \-xarch=v9 \-I/usr/local/include

&nbsp;    Binary compatibility with Perl 5\.005? \[y\] n

&nbsp;    Do you want to install perl as /usr/bin/perl? \[y\] n

&nbsp;    What is your domain name? \[\.inside\.zvolve\.net\] \.zvolve\.com

&nbsp;    What is your e\-mail address? \[garry@&#8203;exper\.zvolve\.com\] garry@&#8203;zvolve\.com

<details><summary>Perl Info</summary>

```

Flags:
    category=install
    severity=low

Site configuration information for perl v5.6.0:

Configured by garry at Sat Jun 17 10:27:41 EDT 2000.

Summary of my perl5 (revision 5.0 version 6 subversion 0) configuration:
  Platform:
    osname=solaris, osvers=2.7, archname=sun4-solaris
    uname='sunos exper 5.7 generic_106541-11 sun4u sparc sunw,ultra-60 '
    config_args='-Dprefix=/home/garry/perl'
    hint=recommended, useposix=true, d_sigaction=define
    usethreads=undef use5005threads=undef useithreads=undef usemultiplicity=undef
    useperlio=undef d_sfio=undef uselargefiles=define 
    use64bitint=define use64bitall=define uselongdouble=undef usesocks=undef
  Compiler:
    cc='cc', optimize='-O', gccversion=
    cppflags='-xarch=v9 -I/usr/local/include'
    ccflags ='-xarch=v9 -I/usr/local/include -D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64 -xarch=v9 '
    stdchar='char', d_stdstdio=undef, usevfork=false
    intsize=4, longsize=8, ptrsize=8, doublesize=8
    d_longlong=define, longlongsize=8, d_longdbl=define, longdblsize=16
    ivtype='long', ivsize=8, nvtype='double', nvsize=8, Off_t='off_t', lseeksize=8
    alignbytes=8, usemymalloc=y, prototype=define
  Linker and Libraries:
    ld='cc', ldflags =' -L/usr/local/lib  -xarch=v9'
    libpth=/usr/lib/sparcv9 /usr/local/lib
    libs=-lsocket -lnsl -ldl -lm -lc -lsec
    libc=/usr/lib/sparcv9/libc.so, so=so, useshrplib=false, libperl=libperl.a
  Dynamic Linking:
    dlsrc=dl_dlopen.xs, dlext=so, d_dlsymun=undef, ccdlflags=' '
    cccdlflags='-KPIC', lddlflags=' -G -xarch=v9 -L/usr/local/lib'

Locally applied patches:
    


@INC for perl v5.6.0:
    /home/garry/perl/lib/5.6.0/sun4-solaris
    /home/garry/perl/lib/5.6.0
    /home/garry/perl/lib/site_perl/5.6.0/sun4-solaris
    /home/garry/perl/lib/site_perl/5.6.0
    /home/garry/perl/lib/site_perl
    .


Environment for perl v5.6.0:
    HOME=/home/garry
    LANG (unset)
    LANGUAGE (unset)
    LC_COLLATE=en_US.ISO8859-1
    LC_CTYPE=en_US.ISO8859-1
    LC_MESSAGES=C
    LC_MONETARY=en_US.ISO8859-1
    LC_NUMERIC=en_US.ISO8859-1
    LC_TIME=en_US.ISO8859-1
    LD_LIBRARY_PATH (unset)
    LOGDIR (unset)
    PATH=/home/garry/perl/bin:/bin:/sbin:/usr/sbin:/usr/ccs/bin:/usr/local/bin:/usr/openwin/bin:/usr/java1.1/bin:/usr/dt/bin:/home/garry/bin
    PERL_BADLANG (unset)
    SHELL=/bin/ksh


```
</details>

**Comment by p5pRT** [comment]
### From @smpeters

> \[garry@&#8203;ifr\.inside\.zvolve\.net \- Sat Jun 17 00&#8203;:59&#8203;:47 2000\]&#8203;:
> 
> 
> This is a bug report for perl from garry@&#8203;zvolve\.com\,
> generated with the help of perlbug 1\.28 running under perl v5\.6\.0\.
> 
> 
> \-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-
> \[Please enter your report here\]
> 
> Configure fails when selecting 64\-bit support and using the Sun
> Workshop compiler&#8203;:
> 
> There is an error in hints/solaris\_2\.sh that generates a \`dot' command
> that specifies the wrong path for a file\.  It's on line 417\.  Here is
> a patch that fixes the problem&#8203;:
> 
>     \[exper\] /home/garry/src/perl\-5\.6\.0 $ diff \-c ~/xx
> hints/solaris\_2\.sh
>     \*\*\* /home/garry/xx      Sat Jun 17 10&#8203;:41&#8203;:30 2000
>     \-\-\- hints/solaris\_2\.sh  Sat Jun 17 10&#8203;:41&#8203;:48 2000
>     \*\*\*\*\*\*\*\*\*\*\*\*\*\*\*
>     \*\*\* 414\,421 \*\*\*\*
>       EOM
>             exit 1
>         fi
>     \! \#         \. \./UU/workshoplibpth\.cbu
>     \!           \. \./workshoplibpth\.cbu
>         case "$cc \-v 2>/dev/null" in
>         \*gcc\*\)
>             echo 'main\(\) \{ return 0; \}' > try\.c
>     \-\-\- 414\,420 \-\-\-\-
>       EOM
>             exit 1
>         fi
>     \!           \. \./UU/workshoplibpth\.cbu
>         case "$cc \-v 2>/dev/null" in
>         \*gcc\*\)
>             echo 'main\(\) \{ return 0; \}' > try\.c
>     \[exper\] /home/garry/src/perl\-5\.6\.0 $
> 
> I used the following options to install&#8203;:
> 
>     mkdir ~/perl ~/perl/bin
> 
>     sh Configure \-Dprefix=/home/garry/perl
> 
>     Directories to use for library searches?
>     \[/usr/local/lib /opt/SUNWspro/SC5\.0/lib /lib /usr/lib
> /usr/ccs/lib\] /usr/lib/sparcv9 /usr/local/lib
> 
>     Any additional cc flags? \[\-I/usr/local/include\] \-xarch=v9
> \-I/usr/local/include
> 
>     Binary compatibility with Perl 5\.005? \[y\] n
> 
>     Do you want to install perl as /usr/bin/perl? \[y\] n
> 
>     What is your domain name? \[\.inside\.zvolve\.net\] \.zvolve\.com
> 
>     What is your e\-mail address? \[garry@&#8203;exper\.zvolve\.com\]
> garry@&#8203;zvolve\.com
> 

It appears that the solaris\_2\.sh hints file has long since changed\, so
the patch above doesn't apply anymore\.  Based on evidence from the
various smokes currently performed\, this problem appears to have been
resolved\.

**Comment by p5pRT** [comment]
@smpeters - Status changed from 'open' to 'resolved'
